systemtest/src/test/java/io/strimzi/systemtest/operators/MultipleClusterOperatorsIsolatedST.java

```diff
@@ -205,22 +304,43 @@ void testKafkaCCAndRebalanceWithMultipleCOs(ExtensionContext extensionContext) {
 
         LOGGER.info("Checking if KafkaRebalance is still ignored, after the cluster stability wait");
 
         // because KR is ignored, it shouldn't contain any status
-        assertNull(KafkaRebalanceResource.kafkaRebalanceClient().inNamespace(DEFAULT_NAMESPACE).withName(clusterName).get().getStatus());
+        assertNull(KafkaRebalanceResource.kafkaRebalanceClient().inNamespace(testStorage.getNamespaceName()).withName(testStorage.getClusterName()).get().getStatus());
 
         LOGGER.info("Adding {} selector of {} to Kafka", SECOND_CO_SELECTOR, SECOND_CO_NAME);
-        KafkaResource.replaceKafkaResourceInSpecificNamespace(clusterName, kafka -> kafka.getMetadata().setLabels(SECOND_CO_SELECTOR), DEFAULT_NAMESPACE);
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(), kafka -> kafka.getMetadata().setLabels(SECOND_CO_SELECTOR), testStorage.getNamespaceName());
 
         LOGGER.info("Waiting for Kafka to scales pods to {}", scaleTo);
-        RollingUpdateUtils.waitForComponentAndPodsReady(DEFAULT_NAMESPACE, kafkaSelector, scaleTo);
+        RollingUpdateUtils.waitForComponentAndPodsReady(testStorage.getNamespaceName(), kafkaSelector, scaleTo);
+
+        assertThat(PodUtils.podSnapshot(testStorage.getNamespaceName(), kafkaSelector).size(), is(scaleTo));
+
+        KafkaRebalanceUtils.doRebalancingProcess(new Reconciliation("test", KafkaRebalance.RESOURCE_KIND, testStorage.getNamespaceName(), testStorage.getClusterName()), testStorage.getNamespaceName(), testStorage.getClusterName());
 
-        assertThat(PodUtils.podSnapshot(DEFAULT_NAMESPACE, kafkaSelector).size(), is(scaleTo));
+        LOGGER.info("Verifying that operands are operated by expected Cluster Operator {}", FIRST_CO_NAME);
+        MetricsUtils.assertMetricResourcesHigherThanOrEqualTo(secondCoMetricsCollector, KafkaRebalance.RESOURCE_KIND, 1);
+        MetricsUtils.assertMetricResourcesHigherThanOrEqualTo(secondCoMetricsCollector, Kafka.RESOURCE_KIND, 1);
+    }
+
+    private MetricsCollector setupCOMetricsCollectorInNamespace(String coName, String coNamespace) {
+        // necessary for finding correct deployment when building metric collector.
+        ResourceManager.setCoDeploymentName(coName);
+
+        String coScraperName = coNamespace + "-" + Constants.SCRAPER_NAME;
+        String coScraperPodName = ResourceManager.kubeClient().listPodsByPrefixInName(coNamespace, coScraperName).get(0).getMetadata().getName();
+
+        MetricsCollector clusterOperatorCollector = new MetricsCollector.Builder()
+            .withScraperPodName(coScraperPodName)
+            .withNamespaceName(coNamespace)
+            .withComponentType(ComponentType.ClusterOperator)
+            .withComponentName("")
+            .build();
 
-        KafkaRebalanceUtils.doRebalancingProcess(new Reconciliation("test", KafkaRebalance.RESOURCE_KIND, DEFAULT_NAMESPACE, clusterName), DEFAULT_NAMESPACE, clusterName);
+        return clusterOperatorCollector;
```

Comment:
Wouldn't be better to move it to the `MetricsCollector` or some metrics utils class, pass the scraper name as a parameter, so it can be used in the future by other tests? Just an idea, WDYT?

Reply:
moved to utils. 👍 but i added getting/setting of CO name otherwise this would crash due to not being able to find CO.